pkg/router/template/configmanager/haproxy/converter.go

```diff
@@ -0,0 +1,88 @@
+package haproxy
+
+import (
+	"bytes"
+	"encoding/csv"
+	"io"
+
+	"github.com/gocarina/gocsv"
+	"github.com/golang/glog"
+)
+
+// Converter transforms a set of bytes. The haproxy dynamic API command
+// responses are not always csv/compliant. This allows us to inject custom
+// converters to make the responses valid csv and parseable.
+type Converter interface {
+	// Convert converts a set of bytes.
+	Convert(data []byte) ([]byte, error)
+}
+
+// ByteConverterFunc converts bytes!
+type ByteConverterFunc func([]byte) ([]byte, error)
+
+// CSVConverter is used to convert the haproxy dynamic configuration API
+// responses into something that is valid CSV and then parse the response
+// and unmarshal it into native golang structs.
+type CSVConverter struct {
+	headers       []byte
+	out           interface{}
+	converterFunc ByteConverterFunc
+}
+
+// NewCSVConverter returns a new CSVConverter.
+func NewCSVConverter(headers string, out interface{}, fn ByteConverterFunc) CSVConverter {
```

Comment:
Should be a pointer unless you have a good reason not to.

Reply:
Not really - ok will change.